python/nebula2/data/ResultSet.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+# --coding:utf-8--
+
+# Copyright (c) 2020 vesoft inc. All rights reserved.
+#
+# This source code is licensed under Apache 2.0 License,
+# attached with Common Clause Condition 1.0, found in the LICENSES directory.
+
+from collections import deque
+from nebula2.data.DataObject import ConvertValue
+
+from nebula2.Exception import (
+    OutOfRangeException,
+    InvalidKeyException
+)
+
+
+class Record(object):
+    def __init__(self, values, names):
+        self._record = list()
+        self._record = values
+        self._size = len(self._record)
+        self._names = dict()
+
+        for index, name in enumerate(names, start=0):
+            self._names[name] = index
+
+        assert len(self._names) == self._size
+
+    def __iter__(self):
+        return iter(self._record)
+
+    def get_type(self, index):
+        '''
+        get type by index
+        :return: value type
+        '''
+        if index >= self._size:
+            raise OutOfRangeException()
+        return self._record[index].getType()
+
+    def get_type(self, key):
+        '''
+        get type by key
+        :return: value type
+        '''
+        if key not in self._names:
+            raise InvalidKeyException(key)
+        return self._record[self._names[key]].getType()
+
+    def get_value(self, index):
+        '''
+        get value by index
+        :return: Value
+        '''
+        if index >= self._size:
+            raise OutOfRangeException()
+        return self._record[index]
+
+    def get_value(self, key):
+        '''
+        get value by key
+        :return: Value
+        '''
+        if key not in self._names:
+            raise InvalidKeyException(key)
+        return self._record[self._names[key]]
+
+    def values(self):
+        return self._record
+
+    def get_path(self, index):
+        '''
+        get path by index
+        :return: Path
+        '''
+        if index >= self._size:
+            raise OutOfRangeException()
+        return self._convert(self._record[index])
```

Comment:
What if the self._record[index] is not of type path? Func convert() may need to check the type at first

Reply:
The `ConvertValue.convert` has do it.

Reply:
The user needs to judge the type is `path`, then it can call get_path(), but here it is better to check, if not return empty.